### DISCUSSION TITLE
EFX syntax for sets intersection

### DISCUSSION THREAD

**Discussion by eduardsmirnov** [OP-TED/eForms-SDK#305] [Q&A]
SDK 1.5, fields.json, assert for field "BT-262-Lot"
has condition 
`"condition" : "{ND-LotProcurementScope} ${(OPP-070-notice in ('1','4','7','8','9','10','12','16','17','18','19','20','23','25','29','33','36','38')) and (BT-11-Procedure-Buyer in ('org-sub', 'org-sub-cga', 'org-sub-ra', 'org-sub-la')) and (BT-262-Lot is present)}",
`

If I understand it right, BT-11-Procedure-Buyer may contain many values if we have many buyers, so it is physically a set of values.
A list of compared values ('org-sub', 'org-sub-cga' ...) is also physically a set of values.

it means that the condition above wants to compare one set with another, or in other words, find an intersection of two sets. But it is not reflected in the syntax. 

Do you think it could be reflected correctly in the syntax? From your example EfxExpressionTranslatorV1.java I saw the handling of exitIntersectFunction() which probably is needed in this context but it is not used.

**Comment by YvesJo** [comment]
Just for clarification:
- there may be multiple buyers in a notice,
- for a given buyer, there may only be one occurrence of BT-11-Procedure-Buyer,
- each occurrence of BT-11-Procedure-Buyer may only contain a single value.

**Reply by eduardsmirnov** [reply]
I understand.
But we are in the context of BT-262-Lot. There are many buyers.
BT-11-Procedure-Buyer may give me many values, e.g. ('org-sub-ra', 'org-sub-la')
How should I read from the current syntax that there is an intersection of two sets?
e.g. in this assert case it looks like ('org-sub-ra', 'org-sub-la') in ('org-sub', 'org-sub-cga', 'org-sub-ra', 'org-sub-la') 

I am not working with XPath but with database.
I really need a clear difference in EFX between "single value in a set of values" and "set of values intersects with a set of values"
